nautobot/extras/forms.py

```diff
@@ -1207,6 +1207,32 @@ class RelationshipAssociationFilterForm(BootstrapMixin, forms.Form):
     )
 
 
+class RelationshipAssociationModelFilterForm(forms.Form):
+    def __init__(self, *args, **kwargs):
+        self.obj_type = ContentType.objects.get_for_model(self.model)
+        relationships = Relationship.objects.filter(source_type=self.obj_type)
+        super().__init__(*args, **kwargs)
+        for relationship in relationships:
+            app, model = str(relationship.destination_type).lower().split(" | ")
+            if relationship.source_label and relationship.destination_label:
+                field_name = "Destination for {} to {} relationship".format(
+                    relationship.source_label, relationship.destination_label
+                )
+            else:
+                field_name = "Destination for {} relationship".format(relationship.slug)
+            destination_model = relationship.destination_type.model_class()
+            self.fields[field_name] = DynamicModelMultipleChoiceField(
+                display_field="name",
+                to_field_name="id",
+                queryset=destination_model.objects.all().order_by("name"),
+                required=False,
+                label=f"{field_name}",
+                widget=APISelectMultiple(
+                    api_url=f"/api/{app}/{model}s/",
+                ),
```

Comment:
Do we not get the correct widget for free by using `DynamicModelMultipleChoiceField` with the correct `queryset`?